sycl/test-e2e/GroupAlgorithm/root_group.cpp

```diff
@@ -0,0 +1,116 @@
+}
+
+void testQueriesAndProperties() {
+  sycl::queue q;
+  const auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  const auto kernel = bundle.get_kernel<class QueryKernel>();
+  const auto maxWGs = kernel.ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  const auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+  q.single_task<class QueryKernel>(props, []() {});
+  static_assert(std::is_same_v<std::remove_cv<decltype(maxWGs)>::type, size_t>,
+                "max_num_work_group_sync query must return size_t");
+  assert(maxWGs >= 1 && "max_num_work_group_sync query failed");
+}
+
+void testRootGroup() {
+  sycl::queue q;
+  const auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  const auto kernel = bundle.get_kernel<class RootGroupKernel>();
+  const auto maxWGs = kernel.ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  const auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+
+  int *data = sycl::malloc_shared<int>(maxWGs * WorkGroupSize, q);
+  const auto range = sycl::nd_range<1>{maxWGs * WorkGroupSize, WorkGroupSize};
+  q.parallel_for<class RootGroupKernel>(range, props, [=](sycl::nd_item<1> it) {
+    auto root = it.ext_oneapi_get_root_group();
+    data[root.get_local_id()] = root.get_local_id();
+    sycl::group_barrier(root);
+
+    root = sycl::ext::oneapi::experimental::this_kernel::get_root_group<1>();
+    int sum = data[root.get_local_id()] +
+              data[root.get_local_range() - root.get_local_id() - 1];
+    sycl::group_barrier(root);
+    data[root.get_local_id()] = sum;
+  });
+  q.wait();
+
+  const int workItemCount = static_cast<int>(range.get_global_range().size());
+  for (int i = 0; i < workItemCount; i++) {
+    assert(data[i] == (workItemCount - 1));
+  }
+}
+
+void testRootGroupFunctions() {
+  sycl::queue q;
+  const auto bundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(q.get_context());
+  const auto kernel = bundle.get_kernel<class RootGroupFunctionsKernel>();
+  const auto maxWGs = kernel.ext_oneapi_get_info<
+      sycl::ext::oneapi::experimental::info::kernel_queue_specific::
+          max_num_work_group_sync>(q);
+  const auto props = sycl::ext::oneapi::experimental::properties{
+      sycl::ext::oneapi::experimental::use_root_sync};
+
+  constexpr int testCount = 10;
+  bool *testResults = sycl::malloc_shared<bool>(testCount, q);
+  const auto range = sycl::nd_range<1>{maxWGs * WorkGroupSize, WorkGroupSize};
+  q.parallel_for<class RootGroupFunctionsKernel>(
+      range, props, [=](sycl::nd_item<1> it) {
+        const auto root = it.ext_oneapi_get_root_group();
+        if (root.leader()) {
```

Comment:
Nit: Should we be testing some of the other work-items, too?  Your implementation of the class looks fine, but if I'm reading this test correctly it would pass for incorrect implementations (e.g. if `root.get_local_id()` always returned 0).